FormalConjectures/ForMathlib/Combinatorics/Basic.lean

```diff
@@ -89,32 +85,55 @@ theorem IsSidon.insert {A : Set α} {m : α} [IsRightCancelAdd α] [IsLeftCancel
     · simp_all
       exact fun _ _ _ _ _ ↦ by simp_all [add_comm]
 
+end Set
+
+namespace Finset
+
+-- TODO: remove once https://github.com/leanprover-community/mathlib4/pull/28241 is merged
+@[simp, push_cast]
+theorem isSidon_toSet {A : Finset α} : IsSidon A.toSet ↔ IsSidon A := by
+  simp [IsSidon]
+
+instance (A : Finset α) [DecidableEq α] : Decidable (IsSidon A) :=
+  decidable_of_iff (∀ᵉ (i₁ ∈ A) (j₁ ∈ A) (i₂ ∈ A) (j₂ ∈ A), _) <| by rfl
+
+/-- The maximum size of a Sidon set in the supplied `Finset`. -/
+def IsSidon.maxSubsetCard (A : Finset α) [DecidableEq α] : ℕ :=
```

Comment:
Small nitpick: I'm not sure the `IsSidon` namespace is the best here since `IsSidon` doesn't appear in the hypotheses. Perhaps the following is better?
```suggestion
def maxSidonSubsetCard (A : Finset α) [DecidableEq α] : ℕ :=
```